feat: Ability for user to add/edit/delete parcels (BETA)

jmdevita opened the pull request:
This is an initial state of having the user add parcels through home assistant. I reverse-engineered how the web UI works. So while this is an initial test with just adding parcels, I should be able to add edit/delete fairly easily now. It's a little hack-y to get the account_token, but the account token looks to last for over a year so it's essentially a one-time add.

Will add more logic and tests for this, but wanted to get the initial push out for testing!

jmdevita commented:
@PineappleEmperor wanted to get this over to you! Obv very early stages, but I got it to a point where it works (no tests built, something is failing anyways, lack of clear directions outside of the Readme, etc. etc.)

PineappleEmperor commented:
Amazing! I'll do some testing :) Are you thinking we merge this now or keep it in draft for a time until it's complete and stable?

jmdevita commented:
Few more things I want to clean up from a UI side, but it's almost ready to merge! :)

PineappleEmperor commented:
Sorry for the delay - works beautifully in my testing! The only comment I'd make is the text tip around the courier codes - many of them are larger than 3 characters and we may just need to point at the list: https://api.parcel.app/external/supported_carriers.json.

I wonder if we could add some kind of daily check on the cookie to determine it's still live. It should be immediately obvious from logs, but nicer if we could raise a HASS notification for a 'fix'.

Great job all in all though! Happy to merge when you're ready :) Feels like it'd be a bump to 2.0.0!

jmdevita commented:
No rush @PineappleEmperor! I'm pretty busy until Thursday, so I'll work to add those in. You're right though, I'll think through a way to easily link the carrier (even if it's just a link to that page to start)

PineappleEmperor commented:
Life gets in the way of HASS 😄
On the carrier stuff - I think a nod would be enough to be honest; anyone using the service _should_ expect to have to be precise with the service call. I'm going to test it with a few IMAP/regex automations to template add deliveries automatically.

jmdevita commented:
New updates:
- Added ability to edit and delete parcels
- Added to translation file to point to the keys in supported_carriers.json
- Updated readme

Future additions
- Move courier section to a dropdown list instead (needs to be dynamically updated)
- Tests? Not sure how to implement tests for this one, but will noodle on it

PineappleEmperor commented:
Sounds good to me! Punch it! I THINK one of the new actions should create a draft release after as well if you want to check it out and then publish? :)

jmdevita commented:
Lol so dumb question... how do I check this? @PineappleEmperor ? Is it the Release Drafter action?

PineappleEmperor commented:
Haha, not dumb! What was dumb was my first attempt to get it working 😅

You're spot on - when you merge the pull request it'll (supposedly) draft a release that you can find under releases; from there you can publish the release.

jmdevita commented:
Ahhhh got it! Will merge and look there then :)
